src/data/move.ts

```diff
@@ -3993,12 +3993,34 @@ export class FriendshipPowerAttr extends VariablePowerAttr {
   }
 }
 
-export class HitCountPowerAttr extends VariablePowerAttr {
+/**
+ * This Attribute calculates the current power of {@linkcode Moves.RAGE_FIST}
+ * The counter for power calculation does not reset on every wave but on every new arena encounter
+ */
+export class RageFistPowerAttr extends VariablePowerAttr {
   apply(user: Pokemon, target: Pokemon, move: Move, args: any[]): boolean {
-    (args[0] as Utils.NumberHolder).value += Math.min(user.battleData.hitCount, 6) * 50;
+    const BDHitCount = user.battleData.hitCount;
+    const previousHitCount = user.battleData.prevHitCount;
```

Comment:
```suggestion
    const { hitCount, prevHitCount } = user.battleData;
    const basePower: Utils.NumberHolder = args[0];
```
There's a shortcut you can do to assign variables from an object: `const { varName } = object;`
So in this case, you can grab `hitCount` and `prevHitCount` from `user.battleData` like this.

You can do a similar thing with arrays. For example:
```ts
const myArray = [0, 1, 2];
const [ zeroVal, oneVal, twoVal ] = myArray;

console.log(zeroVal); // 0
console.log(oneVal); // 1
console.log(twoVal); // 2
```
cf https://www.typescriptlang.org/docs/handbook/variable-declarations.html#destructuring